docs/resources/server/enchantments/builtin.md

```diff
@@ -0,0 +1,217 @@
+# Built-In Enchantment Effect Components
+
+Vanilla Minecraft provides numerous different types of enchantment effect components for use in [enchantment] definitions. This article will explain each, including their usage and in-code definition.
+
+## Value Effect Components
+
+_See also [Value Effect Components] on the Minecraft Wiki_
+
+Value effect components are used for enchantments that alter a numerical value somewhere in the game, and are implemented by the class `EnchantmentValueEffect`. 
+
+Value effect components can be set to use any of these operations on their given values:
+- `minecraft:set`: Overwrites the given level-based value.
+- `minecraft:add`: Adds the specified level-based value to the old one.
+- `minecraft:multiply`: Multiplies the specified level-based factor by the old one.
+- `minecraft:remove_binomial`: Polls a given (level-based) chance using a binomial distibution. If it works, subtracts 1 from the value. Note that many values are effectively flags, being fully on at 1 and fully off at 0.
+- `minecraft:all_of`: Accepts a list of other value effects and applies them in the stated sequence.
+
+The Sharpness enchantment uses `minecraft:damage`, a value effect component, as follows to achieve its effect:
+
+```json5
+"effects": {
+  "minecraft:damage": [
+    {
+      "effect": {
+        "type": "minecraft:add",
+        "value": {
+          "type": "minecraft:linear",
+          "base": 1.0,
+          "per_level_above_first": 0.5
+        }
+      }
+    }
+  ]
+}
```

Comment:
Add comments for each part to give a better idea of what's going on. Also, show a datagen tab version.

Reply:
This isn't a complete example of a data generated file, so how would I show data generation for it? I could show datagen that makes the entire Sharpness enchantment, but then it would feel out of scope for this section, wouldn't it?

Reply:
You could just have a comment at the top that says something like `// Passed for 'effects' in an Enchantment` and then show the relevant object.